src/test/cursor.rs

```diff
@@ -0,0 +1,86 @@
+use std::time::Duration;
+
+use bson::doc;
+use futures::{future::Either, StreamExt};
+
+use crate::{
+    options::{CreateCollectionOptions, CursorType, FindOptions},
+    test::{TestClient, LOCK},
+    RUNTIME,
+};
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn tailable_cursor() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    client
+        .drop_collection(function_name!(), function_name!())
+        .await;
+
+    let db = client.database(function_name!());
+
+    db.create_collection(
+        function_name!(),
+        CreateCollectionOptions::builder()
+            .capped(true)
+            .max(5)
+            .size(1_000_000)
+            .build(),
+    )
+    .await
+    .unwrap();
+
+    let coll = db.collection(function_name!());
+
+    coll.insert_many((0..5).map(|i| doc! { "_id": i }), None)
+        .await
+        .unwrap();
+
+    let await_time = Duration::from_millis(500);
+
+    let mut cursor = coll
+        .find(
+            None,
+            FindOptions::builder()
+                .cursor_type(CursorType::TailableAwait)
+                .max_await_time(await_time)
+                .build(),
+        )
+        .await
+        .unwrap();
+
+    for i in 0..5 {
+        assert_eq!(
+            cursor.next().await.transpose().unwrap(),
+            Some(doc! { "_id": i })
+        );
+    }
+
+    let delay = RUNTIME.delay_for(await_time);
+    let next_doc = cursor.next();
+
+    let next_doc = match futures::future::select(Box::pin(delay), Box::pin(next_doc)).await {
```

Comment:
I think `RUNTIME.timeout` can be used for these.

Reply:
`RUNTIME.timeout` doesn't give us back the non-completed future in the case of a timeout. The idea here is to keep reusing the future rather than calling `next` again so that we can test that it will complete with the right value once a new document is inserted.